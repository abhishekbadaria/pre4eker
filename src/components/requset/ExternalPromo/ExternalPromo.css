.inline {
  display: flex;
  margin: 0 -.3em 0 -.3em;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
}

.buttonLine {
  display: flex;
  margin-bottom: .5rem;
  flex-shrink: 0;
  justify-content: space-between;
}

.h2 {
  margin: 0;
}

.h4 {
  margin: 0;
}