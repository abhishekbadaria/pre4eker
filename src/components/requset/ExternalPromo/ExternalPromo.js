import React from "react";
import PropTypes from "prop-types";

import Input from "src/components/Input/Input";
import Button from "src/components/Button/Button";

import style from "./ExternalPromo.css";

import arrayFunctions from "../arrayFunctions";

const ExternalPromo = ({ externalPromos, setExternalPromo, type }) => {
  return (
    <div className={style.externalPromo}>
      <div className={`${style.inline}`}>
        <h4 className={style.h4}>Внешние промоакции</h4>
        <Button
          action={() =>
            setExternalPromo(arrayFunctions.addItem(externalPromos))
          }
          type="ADD"
        />
      </div>

      {externalPromos.map((item) => (
        <div key={item.number} className={`${style.item}`}>
          <div className={`${style.inline}`}>
            <div className={`${style.inline} ${style.promoTypeBtns}`}>
              <Button
                type="TEXT"
                size="sizeAuto"
                active={item.type === "external"}
                action={() => {
                  const upDatedValue = {
                    ...item,
                    type: "external",
                  };
                  setExternalPromo(
                    arrayFunctions.updateItem(externalPromos, upDatedValue)
                  );
                }}
              >
                Внешняя
              </Button>
              <Button
                type="TEXT"
                size="sizeAuto"
                active={item.type === "mindbox"}
                action={() => {
                  const upDatedValue = {
                    ...item,
                    type: "mindbox",
                  };
                  setExternalPromo(
                    arrayFunctions.updateItem(externalPromos, upDatedValue)
                  );
                }}
              >
                Mindbox
              </Button>
            </div>
            <Button
              action={() => {
                setExternalPromo(
                  arrayFunctions.removeItem(externalPromos, item)
                );
              }}
              type="REMOVE"
            />
          </div>
          <div className={style.inline}>
            <div className={style.half}>
              <Input
                label="ИД"
                name={`externalPromo-${type}-${item.number}-id`}
                value={item.id}
                onChange={(e) => {
                  const upDatedValue = {
                    ...item,
                    id: e.target.value,
                  };
                  setExternalPromo(
                    arrayFunctions.updateItem(externalPromos, upDatedValue)
                  );
                }}
              />
            </div>
            <div className={style.half}>
              <Input
                label="Сумма"
                name={`externalProm-${type}-${item.number}-value`}
                value={item.value}
                onChange={(e) => {
                  const upDatedValue = {
                    ...item,
                    value: e.target.value,
                  };
                  setExternalPromo(
                    arrayFunctions.updateItem(externalPromos, upDatedValue)
                  );
                }}
              />
            </div>
          </div>
        </div>
      ))}
    </div>
  );
};

ExternalPromo.propTypes = {
  externalPromos: PropTypes.array,
  setExternalPromo: PropTypes.func,
  type: PropTypes.string,
};

ExternalPromo.defaultProps = {
  externalPromos: [],
  setExternalPromo: () => ({}),
  type: "",
};
export default ExternalPromo;
