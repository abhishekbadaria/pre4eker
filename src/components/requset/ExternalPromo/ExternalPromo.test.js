import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import ExternalPromo from "./ExternalPromo";

describe("ExternalPromo ", () => {
  describe("with empty body", () => {
    const body = {};
    const setBody = jest.fn();
    const typeOfParrent = "order";

    it("should render only add btn ", () => {
      const { container } = render(
        <ExternalPromo
          body={body}
          setBody={setBody}
          typeOfParrent={typeOfParrent}
        />
      );
      expect(container).toMatchSnapshot();
    });

    it("should handle click on add btn ", () => {
      render(
        <ExternalPromo
          body={body}
          setBody={setBody}
          typeOfParrent={typeOfParrent}
        />
      );
      fireEvent.click(screen.getByRole("button"));
      expect(setBody).toHaveBeenCalledTimes(1);
    });

    it("should render 1 element after click on add btn ", () => {
      const { container } = render(
        <ExternalPromo
          body={body}
          setBody={setBody}
          typeOfParrent={typeOfParrent}
        />
      );
      fireEvent.click(screen.getByRole("button"));
      expect(container).toMatchSnapshot();
    });
  });

  describe("with 1 promo", () => {
    const body = {
      requestedPromotions: [
        {
          type: "disCountType",
          promotion: {
            ids: {
              externalId: 1,
            },
            type: "mindbox",
          },
          amount: 1,
        },
      ],
    };
    const setBody = jest.fn();
    const typeOfParrent = "order";

    it("should render 1 element", () => {
      const { container } = render(
        <ExternalPromo
          body={body}
          setBody={setBody}
          typeOfParrent={typeOfParrent}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
