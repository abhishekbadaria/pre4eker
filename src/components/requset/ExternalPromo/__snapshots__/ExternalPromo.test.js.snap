// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalPromo  with 1 promo should render 1 element 1`] = `
<div>
  <div
    class="externalPromo"
  >
    <div
      class="inline"
    >
      <h4
        class="h4"
      >
        Внешние промоакции
      </h4>
      <button
        class="medium button normal false add"
        type="button"
      >
        <div>
          <svg
            fill="none"
            height="6"
            viewBox="0 0 7 6"
            width="7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.06 3.16H3.828V5.344H3.072V3.16H0.84V2.452H3.072V0.256H3.828V2.452H6.06V3.16Z"
              fill="#000"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="item"
    >
      <div
        class="inline"
      >
        <div
          class="inline promoTypeBtns"
        >
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Внешняя
          </button>
          <button
            class="sizeAuto button normal active basic"
            type="button"
          >
            Mindbox
          </button>
        </div>
        <div
          class="inline promoTypeBtns"
        >
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Скидка
          </button>
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Скидка на доставку
          </button>
        </div>
        <button
          class="medium button normal false remove"
          type="button"
        >
          <div>
            <svg
              fill="none"
              height="1"
              viewBox="0 0 7 1"
              width="7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0 1.90731e-05L6.99999 0L7 0.999981L5.91125e-06 1L0 1.90731e-05Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="inputItem "
          >
            <label
              class="label"
              for="externalPromo-order-0-id"
            >
              ИД акции
            </label>
            <input
              class="input"
              name="externalPromo-order-0-id"
              type="text"
              value="1"
            />
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="inputItem "
          >
            <label
              class="label"
              for="externalProm-order-0-value"
            >
              Сумма
            </label>
            <input
              class="input"
              name="externalProm-order-0-value"
              type="text"
              value="1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExternalPromo  with empty body should render 1 element after click on add btn  1`] = `
<div>
  <div
    class="externalPromo"
  >
    <div
      class="inline"
    >
      <h4
        class="h4"
      >
        Внешние промоакции
      </h4>
      <button
        class="medium button normal false add"
        type="button"
      >
        <div>
          <svg
            fill="none"
            height="6"
            viewBox="0 0 7 6"
            width="7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.06 3.16H3.828V5.344H3.072V3.16H0.84V2.452H3.072V0.256H3.828V2.452H6.06V3.16Z"
              fill="#000"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="item"
    >
      <div
        class="inline"
      >
        <div
          class="inline promoTypeBtns"
        >
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Внешняя
          </button>
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Mindbox
          </button>
        </div>
        <div
          class="inline promoTypeBtns"
        >
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Скидка
          </button>
          <button
            class="sizeAuto button normal false basic"
            type="button"
          >
            Скидка на доставку
          </button>
        </div>
        <button
          class="medium button normal false remove"
          type="button"
        >
          <div>
            <svg
              fill="none"
              height="1"
              viewBox="0 0 7 1"
              width="7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0 1.90731e-05L6.99999 0L7 0.999981L5.91125e-06 1L0 1.90731e-05Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="inputItem "
          >
            <label
              class="label"
              for="externalPromo-order-1-id"
            >
              ИД акции
            </label>
            <input
              class="input"
              name="externalPromo-order-1-id"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="inputItem "
          >
            <label
              class="label"
              for="externalProm-order-1-value"
            >
              Сумма
            </label>
            <input
              class="input"
              name="externalProm-order-1-value"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExternalPromo  with empty body should render only add btn  1`] = `
<div>
  <div
    class="externalPromo"
  >
    <div
      class="inline"
    >
      <h4
        class="h4"
      >
        Внешние промоакции
      </h4>
      <button
        class="medium button normal false add"
        type="button"
      >
        <div>
          <svg
            fill="none"
            height="6"
            viewBox="0 0 7 6"
            width="7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.06 3.16H3.828V5.344H3.072V3.16H0.84V2.452H3.072V0.256H3.828V2.452H6.06V3.16Z"
              fill="#000"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;
